Add configurable corner radius for macOS quick terminal window

doso opened the pull request:
## Summary

Adds a configurable corner radius option for the macOS quick terminal window, allowing users to customize the rounded corners to match their preferences or other macOS windows.

## Changes

- Added `quick-terminal-corner-radius` configuration option (default: ~~16.0 points~~ unset a.k.a 0)
- Updated `QuickTerminalWindow` to accept and apply configurable corner radius
- Updated `QuickTerminalController` to pass config value from `DerivedConfig`
- Added Swift property `quickTerminalCornerRadius` to `Ghostty.Config`

## Testing

- [x] Tested on macOS with default value (~~`16.0`~~ unset)
- [x] Tested with custom values in config file
- [x] Verified corners render correctly in production build
- [x] Verified default matches standard macOS window corner radius

`quick-terminal-corner-radius = 32`

<img width="1800" height="1169" alt="image" src="https://github.com/user-attachments/assets/87e782ce-035c-43bb-88f2-d9c66689fc5b" />

`quick-terminal-corner-radius = 16` ~~(default)~~

<img width="1800" height="1169" alt="image" src="https://github.com/user-attachments/assets/e67ac8fb-8b71-4dab-9ddd-c69ab8999e3a" />

## Configuration

Users can now set the corner radius in their config file `quick-terminal-corner-radius = 20` 

The default value of ~~`16.0`~~ unset a.k.a 0 points matches the standard macOS window corner radius

## Related

This addresses the visual inconsistency where the quick terminal window had sharp corners while other macOS windows have rounded corners. https://github.com/ghostty-org/ghostty/discussions/8666

doso committed:
Add configurable corner radius for macOS quick terminal window

- Add quick-terminal-corner-radius config option (default: 16.0)
- Make QuickTerminalWindow corner radius configurable
- Update QuickTerminalController to use config value
- Add Swift property to read config from Ghostty.Config

This allows users to customize the corner radius of the quick terminal
window to match their preferences or other macOS windows.

bo2themax reviewed (changes requested):
LGTM, but the default value should be the same as previous versions, so that users doesn’t want this don’t have to change their current config

> Could be changed to 0 or a optional value

doso committed:
Refactor quick terminal corner radius handling

- Update QuickTerminalController to handle optional corner radius from configuration.
- Adjust doc to clarify behavior when corner radius is unset.

This change allows for sharper corners when the corner radius is not specified, making corner radius a opt-in.

doso committed:
fix(quick-terminal): ensure rounded corners are reset when radius is zero

doso commented on macos/Sources/Features/QuickTerminal/QuickTerminalController.swift:
```diff
@@ -143,6 +143,14 @@ class QuickTerminalController: BaseTerminalController {
+            let radius = derivedConfig.quickTerminalCornerRadius,
+            radius > 0
+        {
+            qtWindow.setupRoundedCorners(cornerRadius: CGFloat(radius))
```
I've added an else block to explicitly reset the corner radius to 0 when the config value is nil or 0.

frolvanya commented:
@bo2themax do you know what's the approximate timeline for this PR to be merged and if there're any blockers? (no rush, just curious when it hits the tip release)

mitchellh committed:
some cleanups

mitchellh reviewed (changes requested):
I made some changes and I generally want to accept this, but I'm also seeing the background issue as noted by prior reporters. If you resolve this I will merge.
